File: 0443-string-compression/0443-string-compression.py
class Solution(object):
    def compress(self, chars):
        w=0
        l=0

        for r in range(len(chars)):
            if r==len(chars)-1 or chars[r]!=chars[r+1]:
                chars[w]=chars[l]
                w+=1
                c= r - l+1

                if c>1:
                    for d in str(c):
                        chars[w]=d
                        w+=1
                l=r+1
        return w

        

File: 0443-string-compression/0443-string-compression.java
class Solution {
    public int compress(char[] chars) {
        int index=0;
        int count=1;

        for(int i=1;i<=chars.length;i++){
            if(i<chars.length&&chars[i]==chars[i-1]){
                count++;
            }else{
                chars[index++]=chars[i-1];
                if(count>1){
                    String nums=String.valueOf(count);
                    for(char ch:nums.toCharArray()){
                        chars[index++]=ch;
                    }
                }
                count=1;
            }
        }
        return index;
        
    }
}